api/dbadapters/presto.ts

```diff
@@ -0,0 +1,228 @@
+          query: validationQueriesWithoutWrappers[index].query
+        })
+      );
+    }
+    return queryEvaluations;
+  }
+
+  public async catalogs(): Promise<string[]> {
+    const result = await prestoExecute(this.client, "show catalogs");
+    const catalogs = flatten(result.data);
+    return catalogs;
+  }
+
+  public async schemas(): Promise<string[]> {
+    let schemas: string[] = [];
+    await Promise.all(
+      (await this.catalogs()).map(async catalog => {
+        const result = await prestoExecute(this.client, `show schemas from ${catalog}`);
+        schemas = schemas.concat(flatten(result.data).map(schema => `${catalog}.${schema}`));
+      })
+    );
+    // TODO: Currently this returns `catalog.schema` rather than just schemas in order to
+    // conform to string return type. Should this be changed? In tables() it is converted to targets.
+    return schemas;
+  }
+
+  public async createSchema(catalog: string, schema: string): Promise<void> {
+    await prestoExecute(this.client, `create schema if not exists ${catalog}.${schema}`);
+  }
+
+  // TODO: This should take parameters to allow for retrieving from a specific catalog/schema.
+  public async tables(): Promise<dataform.ITarget[]> {
+    let tables: dataform.ITarget[] = [];
+    await Promise.all(
+      (await this.schemas()).map(async schemaString => {
+        const [catalog, schema] = schemaString.split(".");
+        const result = await prestoExecute(this.client, `show tables from ${catalog}.${schema}`);
+        tables = tables.concat(
+          flatten(result.data).map(table => ({ database: catalog, schema, table }))
+        );
+      })
+    );
+    return tables;
+  }
+
+  public async table(target: dataform.ITarget): Promise<dataform.ITableMetadata> {
+    let columnsResult: IPrestoExecutionResult;
+    try {
+      columnsResult = await prestoExecute(this.client, `describe ${resolveTarget(target)}`);
+    } catch (e) {
+      // This probably failed because the table doesn't exist, so ignore as the information isn't necessary.
+    }
+    // TODO: Add primitives mapping.
+    // Columns are structured [column name, type (primitive), extra, comment]. This can be
+    // seen under columnsResult.columns; instead they could be dynamically populated using this info.
+    const fields =
+      columnsResult?.data.map(column =>
+        dataform.Field.create({ name: column[0], description: column[3] })
+      ) || [];
+    return dataform.TableMetadata.create({
+      // TODO: Add missing table metadata items.
+      target,
+      fields
+    });
+  }
+
+  public async preview(target: dataform.ITarget, limitRows: number = 10): Promise<any[]> {
+    const { rows } = await this.execute(
+      `select * from ${target.database}.${target.schema}.${target.name} limit ${limitRows}`
+    );
+    return rows;
+  }
+
+  public async persistedStateMetadata(): Promise<dataform.IPersistedTableMetadata[]> {
+    // TODO.
```

Comment:
nit: let's be consistent with the other DB adapters, rm these 'TODO's (which may never get done) and write `// Unimplemented.`